server/src/python/browser/PyrightLanguageProviderBrowser.ts

```diff
@@ -0,0 +1,268 @@
+// Copyright © 2022-2024, SAS Institute Inc., Cary, NC, USA.  All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import {
+  CallHierarchyIncomingCallsParams,
+  CallHierarchyItem,
+  CallHierarchyOutgoingCall,
+  CallHierarchyOutgoingCallsParams,
+  CallHierarchyPrepareParams,
+  CancellationToken,
+  CompletionItem,
+  CompletionList,
+  CompletionParams,
+  Connection,
+  Declaration,
+  DeclarationLink,
+  Definition,
+  DefinitionLink,
+  DidChangeConfigurationParams,
+  DidChangeWatchedFilesParams,
+  DidCloseTextDocumentParams,
+  DidOpenTextDocumentParams,
+  DocumentHighlight,
+  DocumentHighlightParams,
+  DocumentSymbol,
+  DocumentSymbolParams,
+  ExecuteCommandParams,
+  HoverParams,
+  InitializeParams,
+  InitializeResult,
+  LSPAny,
+  Location,
+  PrepareRenameParams,
+  ReferenceParams,
+  RenameParams,
+  ResultProgressReporter,
+  SignatureHelpParams,
+  SymbolInformation,
+  TextDocumentPositionParams,
+  WorkDoneProgressReporter,
+  WorkspaceSymbol,
+  WorkspaceSymbolParams,
+} from "vscode-languageserver";
+import { TextDocument } from "vscode-languageserver-textdocument";
+
+import { IPythonMode } from "pyright-internal-browser/dist/packages/pyright-internal/src/analyzer/sourceFile";
+import { FileSystem } from "pyright-internal-browser/dist/packages/pyright-internal/src/common/fileSystem";
+import { PyrightServer } from "pyright-internal-browser/dist/packages/pyright-internal/src/server";
+
+import { LanguageServiceProvider } from "../../sas/LanguageServiceProvider";
+import { extractPythonCodes } from "../utils";
+
+export class PyrightLanguageProviderBrowser extends PyrightServer {
```

Comment:
looks like the majority part of code in PyrightLanguageProviderBrowser is same to PyrightLanguageProviderNode
any chance to make a platform-independent PyrightLanguageProvider class (instead of only an interface)?